src/inmanta/db/versions/v1.py

```diff
@@ -23,319 +23,851 @@
-
--- Table: public.resources
-CREATE TABLE IF NOT EXISTS public.resource (
-    environment uuid NOT NULL,
-    model integer NOT NULL,
-    resource_id varchar NOT NULL,
-    resource_version_id varchar NOT NULL,
-    agent varchar NOT NULL,
-    last_deploy timestamp,
-    attributes JSONB,
-    attribute_hash varchar,
-    status resourcestate DEFAULT 'available',
-    provides varchar[] DEFAULT array[]::varchar[],
-    PRIMARY KEY(environment, resource_version_id),
-    FOREIGN KEY (environment, model) REFERENCES configurationmodel (environment, version) ON DELETE CASCADE
+
+--
+-- Name: resourcestate; Type: TYPE; Schema: public; Owner: -
+--
+
+CREATE TYPE public.resourcestate AS ENUM (
+    'unavailable',
+    'skipped',
+    'dry',
+    'deployed',
+    'failed',
+    'deploying',
+    'available',
+    'cancelled',
+    'undefined',
+    'skipped_for_undefined',
+    'processing_events'
 );
 
--- Used in:
---   * data. get_resources_for_attribute_hash()
-CREATE INDEX resource_env_attr_hash_index ON resource (environment, attribute_hash);
--- Used in:
---      * data.get_resources_for_version()
---      * data.get_deleted_resources()
--- => Prevent sequential scan through all resources
-CREATE INDEX resource_env_model_agent_index ON resource (environment, model, agent);
--- Used in:
---      * data.get_resources_report()
---      * data.get_latest_version()
---      * data.get_deleted_resources()
--- => Prevent sequential scan through all resources
--- => Prevent sort operation on column model
-CREATE INDEX resource_env_resourceid_index ON resource (environment, resource_id, model DESC);
--- Used in:
---      * data.get_deleted_resources()
--- => Prevent costly search through jsonb structures
-CREATE INDEX resource_attributes_index ON resource USING gin (attributes jsonb_path_ops);
-
--- Table: public.resourceaction
-CREATE TABLE IF NOT EXISTS public.resourceaction (
-    action_id uuid PRIMARY KEY NOT NULL,
-    action resourceaction_type NOT NULL,
-    started timestamp NOT NULL,
-    finished timestamp,
-    messages JSONB[],
-    status resourcestate,
-    changes JSONB DEFAULT '{}'::jsonb,
-    change change,
-    send_event boolean
+
+--
+-- Name: versionstate; Type: TYPE; Schema: public; Owner: -
+--
+
+CREATE TYPE public.versionstate AS ENUM (
+    'success',
+    'failed',
+    'deploying',
+    'pending'
 );
 
--- Used in:
---      * data.get_log()
--- => Prevent sort operation on column started
-CREATE UNIQUE INDEX resourceaction_action_id_started_index ON resourceaction (action_id, started DESC);
--- Used in:
---      * data.purge_logs()
--- => Prevent sequential scan through all resource actions
-CREATE INDEX resourceaction_started_index ON resourceaction (started);
-
--- Table: public.resourceversionid
--- TODO: FK CONSTRAINT???
-CREATE TABLE IF NOT EXISTS public.resourceversionid (
+
+SET default_tablespace = '';
+
+SET default_table_access_method = heap;
```

Comment:
These have not been part of any database migration script up to now. Why is this required?